source/distributed/shared_tria.cc

```diff
@@ -134,6 +135,20 @@ namespace parallel
       this->update_number_cache ();
     }
 
+    template <int dim, int spacedim>
+    void
+    Triangulation<dim, spacedim>::
+    copy_triangulation (const dealii::Triangulation<dim, spacedim> &old_tria)
+    {
+      Assert ((dynamic_cast<const dealii::parallel::distributed::Triangulation<dim,spacedim> *>(&old_tria) == NULL),
+              ExcMessage("Cannot use this function on parallel::distributed::Triangulation."));
+
+      dealii::parallel::Triangulation<dim,spacedim>::copy_triangulation (old_tria);
+      dealii::Triangulation<dim,spacedim>::copy_triangulation (old_tria);
```

Comment:
agree, we probably should revise this function in p::s::Tria and p::d::Tria to make it less confusing. 

Oddly enough the order won't matter as `p::Tria::copy_tria()` only copies mpi communicator if the old_tria is castable to p::Tria. https://www.dealii.org/developer/doxygen/deal.II/tria__base_8cc_source.html#l00060